src/auth/service.py

```
@@ -1,30 +1,146 @@
+    """Google JWKS를 비동기로 가져오고 캐싱합니다."""
+    global _google_jwks_cache, _google_jwks_fetched_at
+    now = time.monotonic()
+    if _google_jwks_cache and (now - _google_jwks_fetched_at) < _JWKS_CACHE_TTL:
+        return _google_jwks_cache
+
+    async with httpx.AsyncClient() as client:
+        response = await client.get(constants.GOOGLE_JWKS_URL)
+        if response.status_code != 200:
+            if _google_jwks_cache:
+                return _google_jwks_cache
+            raise OAuthError("Failed to fetch Google JWKS")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)